src/lib/models/transaction.js

```diff
@@ -196,6 +196,12 @@ class Transaction extends Model {
 
   set volume (volume) {
     validators.assertNullableQuantity(volume);
```

Comment:
All the validation can be rolled into a single validator here:
```
// validators.js
const assertNullableQuantityWithMaximum(value, max) =>
  assert(value, VALID_NULLABLE_QUANTITY.max(max));
```
And we can make use of the chargeElement.volume property which already does billableAnnualQuantity || authorisedAnnualQuantity:
```
// transaction.js
assertNullableQuantityWithMaximum(value, this.chargeElement.volume);
this._volume = volume;
```

Reply:
I did think to use a validator here, I can't remember why I decided against it

Reply:
If the transaction had a matching error, the volume will have been set to `null` at this point, so I have had to find the max from the two quantities